### PR TITLE
Suppress compiler warnings and fix Python bindings

### PR DESCRIPTION
**Pull request by zkingston**
- Some small changes with uninitialized memory to suppress some huge boost graph warnings
- Marking includes as system to not report internal warnings

### CODE REVIEW

File: src/ompl/control/planners/syclop/Syclop.h [hunk 1]
```diff
@@ -330,21 +329,21 @@ namespace ompl
                 }
                 /** \brief The cells of the underlying coverage grid that contain tree motions originating from
                     direct connections along this adjacency */
-                std::set<int> covGridCells;
+                std::set<int> covGridCells = {0};
```
Comment (mamoll) [inline src/ompl/control/planners/syclop/Syclop.h:332]: I think you want to create an empty set, not a singleton set with 0 as its element.
```suggestion
                std::set<int> covGridCells = {};
```

File: src/ompl/control/planners/syclop/Syclop.h [hunk 2]
```diff
@@ -330,21 +329,21 @@ namespace ompl
...
                 /** \brief The source region of this adjacency edge */
-                const Region *source;
+                const Region *source = {0};
```
Comment (mamoll) [inline src/ompl/control/planners/syclop/Syclop.h:334]:
```suggestion
                const Region *source{nullptr};
```

File: src/ompl/control/planners/syclop/Syclop.h [hunk 3]
```diff
@@ -330,21 +329,21 @@ namespace ompl
...
                 /** \brief The target region of this adjacency edge */
-                const Region *target;
+                const Region *target = {0};
```
Comment (mamoll) [inline src/ompl/control/planners/syclop/Syclop.h:336]:
```suggestion
                const Region *target{nullptr};
```

File: src/ompl/control/planners/syclop/Syclop.h [hunk 4]
```diff
@@ -330,21 +329,21 @@ namespace ompl
...
                 /** \brief The cost of this adjacency edge, used in lead computations */
-                double cost;
+                double cost = {0};
```
Comment (mamoll) [inline src/ompl/control/planners/syclop/Syclop.h:338]:
```suggestion
                double cost = {0.};
```

File: src/ompl/control/planners/syclop/Syclop.h [hunk 5]
```diff
@@ -330,21 +329,21 @@ namespace ompl
...
-                int numLeadInclusions;
+                int numLeadInclusions = {0};
                 /** \brief The number of times the low-level tree planner has selected motions from the source region
                     when attempting to extend the tree toward the target region. */
-                int numSelections;
+                int numSelections = {0};
                 /** \brief This value is true if and only if this adjacency's source and target regions both contain
                  * zero tree motions. */
-                bool empty;
+                bool empty = {0};
```
Comment (mamoll) [inline src/ompl/control/planners/syclop/Syclop.h:346]:
```suggestion
                bool empty{false};
```

**Review by mamoll — CHANGES_REQUESTED** [review]
You should bump the minimum boost version required in CMakeLists.txt, README.md, and docs if you remove `external/boost/python`. Just matching the boost .deb version on the oldest supported version of Ubuntu is a reasonable choice.

**Comment by zkingston** [comment]
No, I'll restore external boost - that should be a separate pull.

**Review by mamoll — APPROVED** [review]

**Pull request merged**